### PR TITLE
ARROW-6509: [Java][CI] Upgrade maven-surefire-plugin to version 3.0.0-M3, disable Gandiva JNI unit tests temporarily

### PR DESCRIPTION
**Pull request by wesm**
To see if the CI issues go away.

### CODE REVIEW

**Comment by emkornfield** [comment]
@ursabot build

**Comment by emkornfield** [comment]
@wesm I pushed some commits that I think will work around the brittleness in java tests when forking=0, did the upgrade of the plugin by itself also fail?    I can't reproduce locally on my mac.  I might try resurrecting a linux VM to see if I can reproduce the error there.

File: java/pom.xml [hunk 1]
```diff
@@ -39,7 +39,7 @@
     <dep.fbs.version>1.9.0</dep.fbs.version>
     <dep.flatc.version>1.9.0</dep.flatc.version>
     <arrow.vector.classifier></arrow.vector.classifier>
-    <forkCount>2</forkCount>
+    <forkCount>0</forkCount>
```
Comment (liyafan82) [inline java/pom.xml]: Thanks for fixing this problem. 

IMO, reducing the value of forkCount seems to be the correct direction of solving the problem. 
However, it also reduces the parallelism of running tests. 
The original value (2) is already small enough. I am not sure if we should further reduce it.

**Comment by wesm** [comment]
OK, now we are getting somewhere

```
[INFO] ------------------------------------------------------------------------
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f5641e16e26, pid=28036, tid=0x00007f5684efe700
#
# JRE version: OpenJDK Runtime Environment (8.0_191-b12) (build 1.8.0_191-8u191-b12-2ubuntu0.16.04.1-b12)
# Java VM: OpenJDK 64-Bit Server VM (25.191-b12 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libgandiva_jni.sodbddc505-1c64-4e65-89f1-5ad5bd499246+0xa5ce26]
#
# Core dump written. Default location: /home/travis/build/apache/arrow/java/core or core.28036
#
# An error report file with more information is saved as:
# /home/travis/build/apache/arrow/java/hs_err_pid28036.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
/home/travis/build/apache/arrow/ci/travis_script_gandiva_java.sh: line 33: 28036 Aborted                 (core dumped) $TRAVIS_MVN test -P arrow-jni -pl gandiva -Darrow.cpp.build.dir=$CPP_BUILD_DIR/debug
```

File: java/pom.xml (hunk 1 above)
Comment (wesm) [reply java/pom.xml]: The bug still hasn't been fixed -- reducing the forkCount to 0 simply is allowing us to see the segfault / core dump that's happening...

**Comment by wesm** [comment]
I restored the forkCount=2 and turned off the Gandiva JNI tests so we can get passing CI builds again

**Comment by bkietz** [comment]
If you filed a bug report at http://bugreport.java.com/bugreport/crash.jsp or anywhere else, could you drop a link to that here and/or on jira?

**Comment by wesm** [comment]
I don't think it's a bug in the JVM, it seems likely it's caused by one of the last few Gandiva-related patches 

https://github.com/apache/arrow/commits/master/cpp/src/gandiva

**Pull request closed without merging**